src/test/logical_query_plan/stored_table_node_test.cpp

```diff
@@ -187,8 +199,7 @@ TEST_F(StoredTableNodeTest, FunctionalDependenciesPrunedRightColumnSet) {
   // Prune column "b", which would be part of the right column set in the resulting FD: {a} => {b, c}
   _stored_table_node->set_pruned_column_ids({ColumnID{1}});
 
-  // Although b is not part of the output column expressions, we want to see an FD returned.
-  const auto fd_expected = FunctionalDependency{{_a}, {_b, _c}};
+  const auto fd_expected = FunctionalDependency{{_a}, {_c}};
```

Comment:
I think, we can not do this. Think of the following example:

```
a  b  c
0  0  0  -> {0} => {0, 0}
1  1  0  -> {1} => {1, 0}
```

After pruning b:
```
a  (b)  c
0  (0)  0  -> {0} => {0}
1  (1)  0  -> {1} => {0}
```

I included the FD anyway because it does not break anything. And in case our pruned column comes back / joins our LQP again, we have the FD still in place.

Reply:
I can't follow. Why is it a problem if we remove columns on the dependent side?

About the example: Any chance you are mixing FDs and uniqueness? The existence of an FD does not guarantee uniqueness on the right side. For example, `zip_code -> {city, state}` might be a valid FD, in which `zip_code -> {state}` holds, too.

Reply:
> Any chance you are mixing FDs and uniqueness?

Indeed, you are right. Thanks!